### PR TITLE
Adding static code analysis, adrs, documentation.

### PR DESCRIPTION
**Pull request by greggmarshall**
Adding the composer require steps for static code analysis, haven't added the configuration file(s) and set it up to scan the code on commits.

Also added the relevant Lullabot ADRs to docs/adrs.  We might want to change the deciders from the Lullabot employees listed to our team members.

### CODE REVIEW

File: docs/adrs/20210729-stage-file-proxy.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - devops
+  - locals
+  - drupal
+
+deciders:
+  - Andrew Berry
+  - Darren Petersen
+  - Marcos Cano
+  - Chris Albrecht
+
```
Comment (jrosgiralt) [inline docs/adrs/20210729-stage-file-proxy.md:21]: @greggmarshall as you mention in your notes, let's change the deciders according to our org. 

Can I suggest that you add the following deciders?

- Jordi Ros-Giralt
- Javier Prieto
- Gregg Marshall (if you think it's ok)

Thanks.

**Review by jrosgiralt — APPROVED** [review]
Code looks good and can be committed. Prior to committing, can you update the deciders list? See my note on the list of deciders to use. Thanks.

**Comment by greggmarshall** [comment]
Updated ADRs with suggested deciders, also added Field Group module to support your config, finally updated core and modules to latest versions.  Generally I like to keep core and modules as up to date as possible during development to minimize later surprises.

**Comment by jrosgiralt** [comment]
That's great, thanks for also adding field group module and updating core & modules.

**Comment by jrosgiralt** [comment]
By the way, what composer commands did you use to update core and modules, these?

```
composer update drupal/core "drupal/core-*" --with-all-dependencies
composer update drupal/modulename --with-dependencies
```

**Comment by greggmarshall** [comment]
Exactly.

**Pull request merged**